tools/alignment_tools.py

```diff
@@ -0,0 +1,186 @@
+    
+    def load_alignments(self, alignments_dir):
+        ''' load the data from the alignments file '''
+        alignments_base = os.path.join(alignments_dir, 'alignments')
+        try:
+            self.check_alignments_file(alignments_base)
+            self.alignments = self.read_alignments()
+        except ValueError as v:
+            print(v)
+            sys.exit()
+        except:
+            raise
+
+    def backup_alignments(self):
+        ''' Backup copy of old alignments '''
+        now = datetime.datetime.now().strftime("%Y-%m-%d--%H_%M_%S")
+        backupfile = '{}_{}{}'.format(os.path.splitext(self.alignments_file)[0], now, os.path.splitext(self.alignments_file)[1])
+        print('Backing up old {} to {}'.format(self.alignments_file, backupfile))
+        os.rename(self.alignments_file, backupfile)
+
+    def write_alignments(self):
+        ''' compile the new alignments '''
+        print('Saving alignments to {}'.format(self.alignments_file))
+        if self.alignments_format == 'json':
+            return json.dumps(self.alignments, indent=2)
+        elif self.alignments_format == 'p':
+            return pickle.dumps(self.alignments)
+        elif self.alignments_format in ('yaml', 'yml'):
+            try:
+                import yaml
+            except ImportError:
+                print("You must have PyYAML installed to use YAMLSerializer")
+            return yaml.dump(self.alignments, default_flow_style=False)       
+
+    def save_alignments(self):
+        ''' Backup copy of old alignments and save new alignments '''
+        self.backup_alignments()
+        alignments = open(self.alignments_file, 'w')
+        alignments.write(self.write_alignments())
+        alignments.close()
+
+class Faces(object):
+    ''' Object to hold the faces that are to be swapped out '''
+    def __init__(self, faces_dir):
+        self.faces_dir = faces_dir
+        self.faces = {}
+
+        self.load_faces()
+
+    def check_folder_exists(self):
+        ''' makes sure that the faces folder exists '''
+        if not os.path.isdir(self.faces_dir):
+            raise ValueError ('The folder {} could not be found'.format(self.faces_dir))
+
+    def load_faces(self):
+        ''' Load the face names into list '''
+        print('Loading faces from {}'.format(self.faces_dir))
+        try:
+            self.check_folder_exists()
+            for face in os.listdir(self.faces_dir):
+                face_filename = os.path.splitext(face)[0]
+                index = int(face_filename[face_filename.rindex('_')+1:])
+                original_file = '{}{}'.format(face_filename[:face_filename.rindex('_')], os.path.splitext(face)[1])
+                if self.faces.get(original_file, '') == '':
+                    self.faces[original_file] = [index]
+                else:
+                    self.faces[original_file].append(index)
+        except ValueError as v:
+            print(v)
+            sys.exit()
+        except:
```

Comment:
Redundant except with a raise that may mask the error output/traceback.

Reply:
As above. Was temporary, now gone